templates/flavors/rke2/rke2ControlPlane.yaml

```diff
@@ -30,8 +31,74 @@ spec:
               name: "linode-token-region"
             nodeSelector:
               node-role.kubernetes.io/control-plane: "true"
+    - path: /var/lib/rancher/rke2/server/manifests/cilium-policy.yaml
+      content: |
+        apiVersion: "cilium.io/v2"
+        kind: CiliumClusterwideNetworkPolicy
+        metadata:
+          name: "cluster-traffic"
+        spec:
+          description: "allow cluster traffic"
+          nodeSelector: {}
+          ingress:
+            - fromEntities:
+              - cluster
+            - fromCIDR:
+              - 10.0.0.0/8
+            - fromEntities:
+              - world
+              toPorts:
+                - ports:
+                  - port: "22"
+                  - port: "9345"
+                  - port: "6443"
+    - path: /var/lib/rancher/rke2/server/manifests/rke2-cilium-config.yaml
       owner: root:root
-      path: /var/lib/rancher/rke2/server/manifests/ccm-linode.yaml
+      permissions: "0640"
+      content: |-
+        apiVersion: helm.cattle.io/v1
+        kind: HelmChart
+        metadata:
+          name: cilium
+          namespace: kube-system
+        spec:
+          targetNamespace: kube-system
+          version: ${CILIUM_VERSION:=1.15.0}
+          chart: cilium
+          repo: https://helm.cilium.io/
+          bootstrap: true
+          valuesContent: |-
+            bgpControlPlane:
+              enabled: true
+            routingMode: native
+            kubeProxyReplacement: true
+            ipv4NativeRoutingCIDR: 10.0.0.0/8
+            tunnelProtocol: ""
+            enableIPv4Masquerade: true
+            egressMasqueradeInterfaces: eth0
+            policyAuditMode: ${FW_AUDIT_ONLY:=true}
+            hostFirewall:
+              enabled: true
+            extraConfig:
+              allow-localhost: policy
+            k8sServiceHost: 10.0.0.2
+            k8sServicePort: 6443
```

Comment:
One drawback of not using helmchartproxy here is that this sets the servicehost for cilium to be 10.0.0.2 which is first control plane node and if that goes down, then cilium-operator will start having issues. We were setting this to loadbalancer ip using helmchartproxy, only k3s was one not using loadbalancer ip and I don't remember if it was on purpose or was intentional due to some limit we were hitting.

Reply:
I've addressed this, looks like k3s is the outlier here with the first CP node not getting ready until the CNI is installed and the helmchartproxy won't install until the CP node is ready. still playing around with some solutions here